test/lifecycle_test.dart

```diff
@@ -176,6 +177,85 @@ main() {
         sharedTypeTests(testSnapshotType);
       });
 
+      group('getDerivedStateFromProps returns', () {
+        test('derived state, resulting in an updated instance state value', () {
+          const Map initialDerivedState = const {
+            'initialDerivedState': 'initial',
+          };
+
+          const Map updatedDerivedState = const {
+            'initialDerivedState': 'updated',
+          };
+
+          final Map initialProps = {'getDerivedStateFromProps': (_, __, ___) => initialDerivedState};
+
+          final Map updatedProps = {'getDerivedStateFromProps': (_, __, ___) => updatedDerivedState};
+
+          var mountNode = new DivElement();
+          var instance = react_dom.render(components2.LifecycleTest(initialProps), mountNode);
+
+          LifecycleTestHelper component = getDartComponent(instance);
+
+          expect(
+              component.lifecycleCalls,
+              equals([
+                matchCall('getInitialState'),
+                matchCall('getDerivedStateFromProps', args: [
+                  initialProps..addAll({'defaultProp': 'default', 'children': []}),
+                  {}
+                ]),
+                matchCall('render', state: initialDerivedState),
+                matchCall('componentDidMount', state: initialDerivedState),
+              ].where((matcher) => matcher != null).toList()));
+
+          component.lifecycleCalls.clear();
+
+          instance = react_dom.render(components2.LifecycleTest(updatedProps), mountNode);
+          component = getDartComponent(instance);
+
+          expect(
+              component.lifecycleCalls,
+              equals([
+                matchCall('getDerivedStateFromProps', args: [
+                  updatedProps..addAll({'defaultProp': 'default', 'children': []}),
+                  initialDerivedState
+                ]),
+                matchCall('shouldComponentUpdate'),
+                matchCall('render', state: updatedDerivedState),
+                matchCall('getSnapshotBeforeUpdate'),
+                matchCall('componentDidUpdate', state: updatedDerivedState),
+              ].where((matcher) => matcher != null).toList()));
+        });
+
+        test('null, leaving the existing instance state value intact', () {
+          const Map initialState = const {
+            'foo': 'sudo',
+          };
+          final Map initialProps = unmodifiableMap(
+              {'getDerivedStateFromProps': (_, __, ___) => null, 'getInitialState': (_) => initialState});
```

Comment:
#nit trailing commas would help with formatting here
```suggestion
              {'getDerivedStateFromProps': (_, __, ___) => null, 'getInitialState': (_) => initialState,});
```